In [1]:
#importing necessary libraries
import pandas as pd
import numpy as np
from scipy import stats
from statsmodels.stats.proportion import proportions_ztest

# Data - Cutlets

In [2]:
#Importing the dataset
df1 = pd.read_csv("D:\Study\Excelr DS\Assignments\AS-3-Hypothesis\Cutlets.csv")
df1.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [3]:
#Getting the basic information of the dataset
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


There are 2 columns and 35 entries in the dataset. There are no null values in the dataset, hence we shall continue with the further process.

## Checking the normality of the data

Using the Shapiro Test for checking the normality:

H0 : The data is normal

H1 : The data is not normal

In [4]:
print(stats.shapiro(df1["Unit A"]))
print(stats.shapiro(df1["Unit B"]))

ShapiroResult(statistic=0.9649458527565002, pvalue=0.3199819028377533)
ShapiroResult(statistic=0.9727300405502319, pvalue=0.5224985480308533)


Conclusion : 

Here, the p-value for both the datset is higher than 0.05, hence we do not reject H0 at 5% level of significance, and conclude that the data is normal. 

## Paired t-test 

H0 : There is no significance difference between the diameters of the Cutlets

H1 : There is a significance difference between the diameters of the Cutlets

In [5]:
stats.ttest_ind(df1["Unit A"], df1["Unit B"])

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

Conclusion:

Here, the p-value = 0.4722 > 0.05, hence we do not reject H0 at 5% level of significance and conclude that, there is no significance difference between the diameters of the cutlets.

# Data - LabTat

In [6]:
#Importing the dataset
df2 = pd.read_csv("D:\Study\Excelr DS\Assignments\AS-3-Hypothesis\LabTAT.csv")
df2.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [7]:
#Getting the basic information of the dataset
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Laboratory 1  120 non-null    float64
 1   Laboratory 2  120 non-null    float64
 2   Laboratory 3  120 non-null    float64
 3   Laboratory 4  120 non-null    float64
dtypes: float64(4)
memory usage: 3.9 KB


There are 4 columns and 120 entries in the dataset. There are no null values in the dataset, hence we shall continue with the further process.

## Checking the normality of the data

Using the Shapiro Test for checking the normality:

H0 : The data is normal

H1 : The data is not normal

In [8]:
print(stats.shapiro(df2["Laboratory 1"]))
print(stats.shapiro(df2["Laboratory 2"]))
print(stats.shapiro(df2["Laboratory 3"]))
print(stats.shapiro(df2["Laboratory 4"]))

ShapiroResult(statistic=0.9901824593544006, pvalue=0.5506953597068787)
ShapiroResult(statistic=0.9936322569847107, pvalue=0.8637524843215942)
ShapiroResult(statistic=0.9886345267295837, pvalue=0.4205053448677063)
ShapiroResult(statistic=0.9913753271102905, pvalue=0.6618951559066772)


Conclusion : 

Here, the p-value for all the datset is higher than 0.05, hence we do not reject H0 at 5% level of significance, and conclude that the data is normal. 

## ANOVA test

H0 : There is no significance difference between the TAT of the laboratories.

H1 : There is a significance difference between any pair of TAT of the laboratories.

In [9]:
stats.stats.f_oneway(df2["Laboratory 1"], df2["Laboratory 2"], df2["Laboratory 3"], df2["Laboratory 4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Conclusion : 

Here, the p-value is less than 0.05, hence we reject the H0 at 5% level of significance and conclude that, there is a significance difference between any pair of TAT of the laboratories.

# Data - BuyerRatio

In [10]:
#Importing the dataset
df3 = pd.read_csv("D:\Study\Excelr DS\Assignments\AS-3-Hypothesis\BuyerRatio.csv")
df3.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [11]:
#Getting the basic information of the dataset
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Observed Values  2 non-null      object
 1   East             2 non-null      int64 
 2   West             2 non-null      int64 
 3   North            2 non-null      int64 
 4   South            2 non-null      int64 
dtypes: int64(4), object(1)
memory usage: 208.0+ bytes


There are 5 columns and 2 entries in the dataset. There are no null values in the dataset, hence we shall continue with the further process.

## Chi-Square Test 

H0 : The male-female buyer ratios are similar across the regions

H1 : The male-female buyer ratios are not similar across the regions

In [12]:
stats.chi2_contingency([df3["East"], df3["West"], df3["North"], df3["South"]])

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

Conclusion:

Here the p-value = 0.6603 > 0.05, hence we do not reject H0 at 5% level of significance, and conclude that the male-female buyer ratios are similar across the regions

# Data - Customer Order Form

In [13]:
#Importing the dataset
df4 = pd.read_csv("D:\Study\Excelr DS\Assignments\AS-3-Hypothesis\Costomer+OrderForm.csv")
df4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [14]:
#Getting the basic information of the dataset
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


There are 4 columns and 300 entries in the dataset. There are no null values in the dataset, hence we shall continue with the further process.

## Chi-Square Test 

H0 : The defective percentage do not vary by centre

H1 : The defective percentage vary by centre

In [15]:
stats.chi2_contingency([df4['Phillippines'].value_counts(), df4['Indonesia'].value_counts(), df4['Malta'].value_counts(), df4['India'].value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

Conclusion : 

Here, the p-value = 0.2771 > 0.05, hence we do not reject H0 at 5% level of significance, and conclude that the defective percentage do not vary by centre